src/flows/Onboarding/hooks.ts

```diff
@@ -186,12 +186,25 @@ const useJSONSchemaForm = ({
         const { schema } = modify(jsfSchema, options.jsfModify);
         jsfSchema = schema;
       }
-      const hasFieldValues = Object.keys(fieldValues).length > 0;
-      const employmentField = jsonSchemaToEmployment[form] as keyof Employment;
-      const employmentFieldData = (employment?.[employmentField] ||
-        {}) as Record<string, unknown>;
+      // const hasFieldValues = Object.keys(fieldValues).length > 0;
+      // const employmentField = jsonSchemaToEmployment[form] as keyof Employment;
+      // const employmentFieldData = (employment?.[employmentField] ||
+      //   {}) as Record<string, unknown>;
+
+      const moneyFields = findFieldsByType(jsfSchema.properties || {}, 'money');
+      const moneyFieldsData = moneyFields.reduce<Record<string, number | null>>(
+        (acc, field) => {
+          acc[field] = convertToCents(fieldValues[field]);
+          return acc;
+        },
+        {},
+      );
```

Comment:
This is the actual fix! The remaining code changed in this file is part of the refactor.